### PR TITLE
AM - Tables Inventory - Section : My Title

### PR DESCRIPTION
**Pull request by KevinDurand974**
# Tables Inventory

### Section - My Title

- [x] Title
- [x] # Views
- [x] Directors
- [x] Production Status
- [x] Action

> Missing modal on directors

* * *

From issue : https://github.com/blockframes/blockframes/issues/6724

### CODE REVIEW

File: apps/festival/festival/src/app/dashboard/title/list/list.component.html [hunk 1]
```diff
@@ -31,22 +31,33 @@ <h1>My Titles</h1>
       </button>
     </header>
 
-    <bf-table [source]="result$ | async" pagination="50" useFilter  clickable (rowClick)="goToTitle($event)">
-      <ng-template colRef="title.international" label="Title" let-title sort>
-        <b>{{ title }}</b>
+    <bf-table [source]="result$ | async" pagination="50" useFilter>
+      <ng-template colRef label="Title" let-movie sort>
+        <u (click)="goToTitle(movie)" class="clickable">{{ movie.title.international | maxLength:50 }}</u>
```
Comment (phpgeekfr) [inline apps/festival/festival/src/app/dashboard/title/list/list.component.html:36]: use `<a` tag with routerLink

File: apps/festival/festival/src/app/dashboard/title/list/list.component.ts [hunk 2]
```diff
@@ -67,4 +73,22 @@ export class ListComponent {
...
+    await this.service.update(id, (movie) => ({
+      ...movie,
+      app: {
+        ...movie.app,
+        [this.app]: {
+          ...movie.app[this.app],
+          status: 'archived',
+        },
+      },
+    }));
```
Comment (phpgeekfr) [inline apps/festival/festival/src/app/dashboard/title/list/list.component.ts:91]: you should create an updateStatus() method on service directly and use this here and also libs/movie/src/lib/movie/dashboard/actions/actions.component.ts

File: apps/festival/festival/src/app/dashboard/title/list/list.component.html [hunk 3]
```diff
@@ -31,22 +31,33 @@ <h1>My Titles</h1>
...
+      <ng-template colRef="productionStatus" label="Production Status" let-prod sort>
         {{ prod | toLabel: 'productionStatus' }}
       </ng-template>
       <ng-template colRef="app.festival.status" label="Status" let-status>
         <bf-tag>{{ status | toLabel: 'storeStatus' }}</bf-tag>
       </ng-template>
+
+      <ng-template colRef label="Actions" let-movie>
+        <div class="action-buttons">
+          <button type="button" mat-icon-button (click)="goToTitle(movie)">
```
Comment (phpgeekfr) [inline apps/festival/festival/src/app/dashboard/title/list/list.component.html:50]: use <a tag with routerLink

File: apps/festival/festival/src/app/dashboard/title/list/list.component.html [hunk 4]
```diff
@@ -31,22 +31,35 @@ <h1>My Titles</h1>
...
-    <bf-table [source]="result$ | async" pagination="50" useFilter  clickable (rowClick)="goToTitle($event)">
-      <ng-template colRef="title.international" label="Title" let-title sort>
-        <b>{{ title }}</b>
-      </ng-template>
-      <ng-template colRef="analytics" label="# Views" let-analytics sort>
-        {{ analytics?.length }}
+    <bf-table [source]="result$ | async" pagination="50" useFilter>
+      <ng-template colRef label="Title" let-movie sort>
+        <a [routerLink]="movie.id" [matTooltip]="movie.title.international">
+          <u>{{ movie.title.international | maxLength:50 }}</u>
```
Comment (phpgeekfr) [inline apps/festival/festival/src/app/dashboard/title/list/list.component.html:37]: don't use `<u>` inside `<a>`

**Comment by KevinDurand974** [comment]
Preview
preview

File: libs/movie/src/lib/movie/+state/movie.service.ts [hunk 5]
```diff
@@ -32,8 +34,9 @@ export class MovieService extends CollectionService<MovieState> {
   constructor(
     private authService: AuthService,
     private permissionsService: PermissionsService,
-    private analyticService: AnalyticsService,
-    private orgService: OrganizationService
+    private analyticservice: AnalyticsService,
```
Comment (phpgeekfr) [inline libs/movie/src/lib/movie/+state/movie.service.ts:37]: Why did you removed camelCase on `analyticService` ?

File: libs/movie/src/lib/movie/dashboard/actions/actions.component.ts [hunk 6]
```diff
@@ -67,21 +67,7 @@ export class DashboardActionsShellComponent {
...
-      },
-    }));
-
-    if (message) {
-      this.snackbar.open(message, '', { duration: 4000 });
-    } else {
-      this.snackbar.open(`Title ${storeStatus[status]}.`, '', { duration: 4000 });
-    }
+    await this.movieService.updateStatus(this.movie.id, status);
+    this.snackbar.open(message ? message : `Title ${storeStatus[status]}.`, '', { duration: 4000 });
```
Comment (phpgeekfr) [inline libs/movie/src/lib/movie/dashboard/actions/actions.component.ts:71]: This should also work: 
```suggestion
    this.snackbar.open(message || `Title ${storeStatus[status]}.`, '', { duration: 4000 });
```

File: apps/festival/festival/src/app/dashboard/title/list/list.component.ts [hunk 7]
```diff
@@ -45,7 +49,9 @@ export class ListComponent {
     private service: MovieService,
     private router: Router,
     private route: ActivatedRoute,
+    private snackbar: MatSnackBar,
     private dynTitle: DynamicTitleService,
+    private cdr: ChangeDetectorRef,
```
Comment (phpgeekfr) [inline apps/festival/festival/src/app/dashboard/title/list/list.component.ts:54]: you are not using it

File: apps/festival/festival/src/app/dashboard/title/list/list.component.ts [hunk 8]
```diff
@@ -67,4 +73,9 @@ export class ListComponent {
   resetFilter() {
     this.filter.reset();
   }
+
+  async changeMovieStatus(movie: Movie, status: StoreStatus) {
+    await this.service.updateStatus(movie.id, status);
+    this.snackbar.open(`Title ${status}.`, '', { duration: 4000 });
```
Comment (phpgeekfr) [inline apps/festival/festival/src/app/dashboard/title/list/list.component.ts:79]: if status is `draft`, message will be 'Title draft` ..
Use the same logic as `async updateStatus(status: StoreStatus, message?: string) {` in `libs/movie/src/lib/movie/dashboard/actions/actions.component.ts`

File: apps/festival/festival/src/app/dashboard/title/list/list.component.html [hunk 9]
```diff
@@ -31,22 +31,49 @@ <h1>My Titles</h1>
...
+            <mat-icon svgIcon="visibility"></mat-icon>
+          </a>
+          <a [routerLink]="['/c/o/dashboard/tunnel/movie', movie.id]" mat-icon-button matTooltip="Edit">
+            <mat-icon svgIcon="edit"></mat-icon>
+          </a>
+          <button type="button" mat-icon-button (click)="changeMovieStatus(movie, 'archived')" matTooltip="Archive">
+            <mat-icon svgIcon="archive"></mat-icon>
+          </button>
+        </div>
+      </ng-template>
```
Comment (phpgeekfr) [inline apps/festival/festival/src/app/dashboard/title/list/list.component.html:76]: Not possible to do something like this ?
```suggestion
      <ng-template  colRef label="Actions" let-movie sticky>
        <div class="action-buttons">
          <a [routerLink]="movie.id" mat-icon-button matTooltip="View">
            <mat-icon svgIcon="visibility"></mat-icon>
          </a>
          <a [routerLink]="['/c/o/dashboard/tunnel/movie', movie.id]" mat-icon-button matTooltip="Edit">
            <mat-icon svgIcon="edit"></mat-icon>
          </a>
          <button  *ngIf="movie.status === 'archived' else archive type="button" mat-icon-button (click)="changeMovieStatus(movie, 'draft')" matTooltip="Restore">
            <mat-icon svgIcon="restore"></mat-icon>
          </button>
          <ng-template #archive>
            <button type="button" mat-icon-button (click)="changeMovieStatus(movie, 'archived')" matTooltip="Archive">
            <mat-icon svgIcon="archive"></mat-icon>
          </button>
          </ng-template>
        </div>
      </ng-template>
```

File: libs/movie/src/lib/movie/+state/movie.service.ts (hunk 5 above)
Comment (KevinDurand974) [reply libs/movie/src/lib/movie/+state/movie.service.ts:37]: It was as that in my branch before but when I merge the updated develop was with a `analyticService`
I can change it.

File: apps/festival/festival/src/app/dashboard/title/list/list.component.ts (hunk 7 above)
Comment (KevinDurand974) [reply apps/festival/festival/src/app/dashboard/title/list/list.component.ts:54]: Yeah, I forgot to remove it, my bad

File: apps/festival/festival/src/app/dashboard/title/list/list.component.html (hunk 9 above)
Comment (KevinDurand974) [reply apps/festival/festival/src/app/dashboard/title/list/list.component.html:76]: I will try, that the version of Remco

Comment (KevinDurand974) [reply apps/festival/festival/src/app/dashboard/title/list/list.component.html:76]: Not working

File: apps/festival/festival/src/app/dashboard/title/list/list.component.html [hunk 10]
```diff
@@ -31,22 +31,40 @@ <h1>My Titles</h1>
...
       </ng-template>
-      <ng-template colRef="productionStatus" label="Production Status" let-prod>
+      <ng-template colRef="productionStatus" label="Production Status" let-prod sort>
         {{ prod | toLabel: 'productionStatus' }}
       </ng-template>
       <ng-template colRef="app.festival.status" label="Status" let-status>
         <bf-tag>{{ status | toLabel: 'storeStatus' }}</bf-tag>
       </ng-template>
+
+      <ng-template  colRef label="Actions" let-movie sticky>
```
Comment (phpgeekfr) [inline apps/festival/festival/src/app/dashboard/title/list/list.component.html:50]:
```suggestion
      <ng-template colRef label="Actions" let-movie sticky>
```

File: apps/festival/festival/src/app/dashboard/title/list/list.component.ts [hunk 11]
```diff
@@ -71,4 +77,9 @@ export class ListComponent {
   resetFilter() {
     this.filter.reset();
   }
+
+  async changeMovieStatus(movie: Movie, status: StoreStatus, message?: string) {
```
Comment (phpgeekfr) [inline apps/festival/festival/src/app/dashboard/title/list/list.component.ts:81]: rename to `updateStatus` like in libs/movie/src/lib/movie/dashboard/actions/actions.component.ts

File: apps/festival/festival/src/app/dashboard/title/list/list.component.html [hunk 12]
```diff
@@ -31,22 +31,40 @@ <h1>My Titles</h1>
...
+        <div class="action-buttons">
+          <a [routerLink]="movie.id" mat-icon-button matTooltip="View">
+            <mat-icon svgIcon="visibility"></mat-icon>
+          </a>
+          <a [routerLink]="['/c/o/dashboard/tunnel/movie', movie.id]" mat-icon-button matTooltip="Edit">
+            <mat-icon svgIcon="edit"></mat-icon>
+          </a>
+          <button *ngIf="movie.app[app].status === 'archived' else archive" type="button" mat-icon-button (click)="changeMovieStatus(movie, 'draft')" matTooltip="Restore">
```
Comment (phpgeekfr) [inline apps/festival/festival/src/app/dashboard/title/list/list.component.html:58]:
```suggestion
          <button *ngIf="movie.app[app].status === 'archived' else archive" type="button" mat-icon-button (click)="updateStatus(movie, 'draft', 'Title restored.')" matTooltip="Restore">
```

**Review by phpgeekfr — APPROVED** [review]

**Pull request merged**